packages/framework-core/src/services/graphql/graphql-type-informer.ts

```diff
@@ -41,25 +44,56 @@ export class GraphQLTypeInformer {
   private metadataPropertiesToGraphQLFields(properties: Array<PropertyMetadata>): GraphQLFieldConfigMap<any, any> {
     const fields: GraphQLFieldConfigMap<any, any> = {}
     for (const prop of properties) {
-      fields[prop.name] = { type: this.getGraphQLTypeFor(prop.type) }
+      if (prop.typeInfo.type === Array) {
+        prop.typeInfo.parameters.forEach((param) => {
+          const graphQLPropType = this.getGraphQLTypeFor(param.type)
+
+          if (!this.canFilter(graphQLPropType)) {
+            const properties = getPropertiesMetadata(param.type)
+            this.generateGraphQLTypeFromMetadata({ class: param.type, properties })
+          }
+
+          fields[prop.name] = {
+            type: GraphQLList(graphQLPropType),
+          }
+        })
+      } else {
+        if (!prop.typeInfo.type.prototype) {
+          fields[prop.name] = {
+            type: new GraphQLEnumType({
+              name: inflected.camelize(prop.name),
+              values: this.generateOperationEnumValuesFor(prop.typeInfo.type),
+            }),
+          }
+        } else {
+          fields[prop.name] = { type: this.getGraphQLTypeFor(prop.typeInfo.type) }
+        }
+      }
     }
     return fields
   }
 
+  public canFilter(graphQLType: GraphQLNonInputType): boolean {
+    return graphQLType instanceof GraphQLScalarType && graphQLType != GraphQLJSONObject
```

Comment:
Can't we also filter complex objects?

Reply:
yes, this just checks if it's a primitive type. I will rename the function.